pkg/kudoctl/kudoinit/crd/crds.go

```diff
@@ -106,27 +123,64 @@ func (c Initializer) verifyIsNotInstalled(client v1beta1.CustomResourceDefinitio
 	return nil
 }
 
-func (c Initializer) verifyInstallation(client v1beta1.CustomResourceDefinitionsGetter, crd *apiextv1beta1.CustomResourceDefinition, result *verifier.Result) error {
-	existingCrd, err := client.CustomResourceDefinitions().Get(context.TODO(), crd.Name, v1.GetOptions{})
+func (c Initializer) getCrdForVerify(client v1beta1.CustomResourceDefinitionsGetter, crdName string, result *verifier.Result) (*apiextv1beta1.CustomResourceDefinition, error) {
+	existingCrd, err := client.CustomResourceDefinitions().Get(context.TODO(), crdName, v1.GetOptions{})
 	if err != nil {
 		if os.IsTimeout(err) {
-			return err
+			return nil, err
 		}
 		if kerrors.IsNotFound(err) {
-			result.AddErrors(fmt.Sprintf("CRD %s is not installed", crd.Name))
-			return nil
+			result.AddErrors(fmt.Sprintf("CRD %s is not installed", crdName))
+			return nil, nil
 		}
-		return fmt.Errorf("failed to retrieve CRD %s: %v", crd.Name, err)
+		return nil, fmt.Errorf("failed to retrieve CRD %s: %v", crdName, err)
 	}
-	if existingCrd.Spec.Version != crd.Spec.Version {
-		result.AddErrors(fmt.Sprintf("Installed CRD %s has invalid version %s, expected %s", crd.Name, existingCrd.Spec.Version, crd.Spec.Version))
-		return nil
+	return existingCrd, nil
+}
+
+func (c Initializer) verifyInstallation(client v1beta1.CustomResourceDefinitionsGetter, crd *apiextv1beta1.CustomResourceDefinition, result *verifier.Result) error {
```

Comment:
There a few `verify*` functions in this file, all sounding very similar. Please update the documentation to make it clear when to use which one 😉